pkg/nodes/node.go

```diff
@@ -189,6 +196,71 @@ func (builder *Builder) ExternalIPv4Network() (string, error) {
 	return extNetwork.IPv4, nil
 }
 
+// IsReady check if the Node is Ready.
+func (builder *Builder) IsReady() (bool, error) {
+	if valid, err := builder.validate(); !valid {
+		return false, err
+	}
+
+	glog.V(100).Infof("Verify %s node availability", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return false, fmt.Errorf("%s node object doesn't exist", builder.Definition.Name)
+	}
+
+	for _, condition := range builder.Object.Status.Conditions {
+		if condition.Type == "Ready" {
+			if condition.Status == isTrue {
```

Comment:
I think you should change this to:
```suggestion
			return condition.Status == isTrue, nil
```
err should be reserved for if an actual error occurs